### PR TITLE
Almost always you want `library`, not `require` 

### PR DESCRIPTION
**Pull request by bpbond**
I’d like to _hurt_ the person who named R’s `require` function, which is exactly for when a package is NOT required. There are a few instances of this being used correctly in the PEcAn code, but almost
all can (and should) be changed to `library`—i.e., the code will not run without that package.

### CODE REVIEW

**Comment by bpbond** [comment]
Looks like this is failing because of #1065 - so will hold until that clears.

File: models/maespa/R/model2netcdf.MAESPA.R [hunk 1]
```diff
@@ -84,14 +84,15 @@ model2netcdf.MAESPA <- function(outdir, sitelat, sitelon, start_date, end_date,
     var[[5]] <- ncvar_def("Qle", units = ("W m-2"), dim = list(lat, lon, t), missval = -999, longname = "latent Heat Flux")
     
     ### Output netCDF data
-    nc      <- nc_create(file.path(outdir, paste(y, "nc", sep = ".")), var)
+    nc <- nc_create(file.path(outdir, paste(y, "nc", sep = ".")), var)
     varfile <- file(file.path(outdir, paste(y, "nc", "var", sep = ".")), "w")
-    for (i in seq_along(var)) {
+    for (i in 1:length(var)) {
```
Comment (mdietze) [inline models/maespa/R/model2netcdf.MAESPA.R]: Inadvertent reversion?

File: modules/data.atmosphere/R/load.cfmet.R [hunk 2]
```diff
@@ -15,11 +15,13 @@
 ##' @export
 ##' @author David LeBauer
 load.cfmet <- cruncep_nc2dt <- function(met.nc, lat, lon, start.date, end.date){
+
+    library(lubridate)
+  library(udunits2)
```
Comment (mdietze) [inline modules/data.atmosphere/R/load.cfmet.R]: extra spaces?

File: modules/data.atmosphere/inst/scripts/met.workflow.NARR.R [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
+library(PEcAn.data.atmosphere)
+library(RPostgreSQL)
 
 #--------------------------------------------------------------------------------------------------#
 # Clear old database connections
-for (i in dbListConnections(PostgreSQL())) db.close(i)
-dbparms <- list(driver=driver, user=user, dbname=dbname, password=password, host=host)
+for (i in dbListConnections(PostgreSQL())) db.close(i) {
+  dbparms <- list(driver=driver, user=user, dbname=dbname, password=password, host=host)
+}
```
Comment (mdietze) [inline modules/data.atmosphere/inst/scripts/met.workflow.NARR.R]: I don't this revised code is correct. The loop should be over db.close, not dbparams.

File: modules/data.remote/inst/scripts/Bayesian_curve_fitting.R [hunk 4]
```diff
@@ -2,7 +2,7 @@
 bayes.curve.fit<-function(outpath,coord.set,fia,n.reps,n.chain){
   
 library(rjags)
-require(R2HTML)
+  library(R2HTML)
```
Comment (mdietze) [inline modules/data.remote/inst/scripts/Bayesian_curve_fitting.R]: extra space?

File: visualization/R/worldmap.R [hunk 5]
```diff
@@ -74,7 +74,7 @@ pecan.worldmap <- function(df.in, outfile = NULL,
          width = 44*diff(xlim)/360, height = 34*diff(ylim)/180, 
          units="in", dpi = 100, bg="transparent")
   
-  require(grid)
+    library(grid)
```
Comment (mdietze) [inline visualization/R/worldmap.R]: spaces?

File: models/maespa/R/model2netcdf.MAESPA.R (hunk 1 above)
Comment (bpbond) [inline models/maespa/R/model2netcdf.MAESPA.R]: Yes, thanks.

**Pull request merged**